### PR TITLE
[doc] Update Earl Grey documentation and block diagram

### PR DESCRIPTION
**Pull request by msfschaffner**
This updates the top-level blockdiagram for the Earl Grey design, and reworks the overview description so that it is more compact. The patch also adds some more detail to the feature list.

Signed-off-by: Michael Schaffner <msf@google.com>

### CODE REVIEW

File: hw/top_earlgrey/doc/_index.md [hunk 1]
```diff
@@ -4,78 +4,120 @@ title: "Earl Grey Top Level Specification"
...
+            <li>AES-128/192/256 with ECB/CBC/CFB/OFB/CTR modes</li>
+            <li>HMAC / SHA2-256</li>
+            <li>KMAC / SHA3-224, 256, 384, 512, [c]SHAKE-128, 256</li>
+            <li>Programmable big number accelerator for RSA and ECC (OTBN)</li>
+            <li>Digital wrapper for analog entropy source</li>
+            <li>NIST-compliant cryptographically secure random number generator (CSRNG)</li>
+            <li>Key manager with DICE support</li>
+            <li>Manufacturing life cycle manager</li>
+            <li>Alert handler for handling critical security events</li>
+            <li>Fuse controller with access controls and memory scrambling</li>
```
Comment (tjaychen) [inline hw/top_earlgrey/doc/_index.md:61]: i feel like `otp` is the more generic term for `fuse`

File: hw/top_earlgrey/doc/_index.md [hunk 2]
```diff
@@ -4,78 +4,120 @@ title: "Earl Grey Top Level Specification"
...
+          </ul>
+        </li>
+        <br></br>
+        <li>Security peripherals:
+          <ul>
+            <li>AES-128/192/256 with ECB/CBC/CFB/OFB/CTR modes</li>
+            <li>HMAC / SHA2-256</li>
+            <li>KMAC / SHA3-224, 256, 384, 512, [c]SHAKE-128, 256</li>
+            <li>Programmable big number accelerator for RSA and ECC (OTBN)</li>
+            <li>Digital wrapper for analog entropy source</li>
```
Comment (tjaychen) [inline hw/top_earlgrey/doc/_index.md:56]: hmmm... i wonder if we should remove the digital wrapper line.  
Of if we do want to keep it, we should probably mention the compliant health checks (in the documentation we mention both FIPS and CC).

**Review by tjaychen — APPROVED** [review]

**Review by arunthomas — APPROVED** [review]
Looks great to me

File: hw/top_earlgrey/doc/_index.md (hunk 1 above)
Comment (msfschaffner) [reply hw/top_earlgrey/doc/_index.md]: Makes sense - changed!

**Pull request merged**